sdks/python/apache_beam/io/iobase.py

```diff
@@ -659,6 +675,167 @@ def fraction_consumed(self):
+    gives the number of split points consumed by the ``BoundedSource`` and the
+    number of split points remaining within the range of the ``RangeTracker``
+    that has not been consumed by the ``BoundedSource``.
+
+    More specifically, given that the position of the current record being read
+    by ``BoundedSource`` is current_position this method produces a tuple that
+    consists of
+    (1) number of split points in the range [self.start_position(),
+    current_position) without including the split point that is currently being
+    consumed. This represents the total amount of parallelism in the consumed
+    part of the source.
+    (2) number of split points within the range
+    [current_position, self.stop_position()) including the split point that is
+    currently being consumed. This represents the total amount of parallelism in
+    the unconsumed part of the source.
+
+    Similar to other methods of class ``RangeTracker``, this method may get
+    invoked by multiple threads hence must be thread safe, e.g. by using a
+    single lock object.
+
+    ** Determining consumed and remaining number of split points **
+
+    Consider the following examples: (1) An input that can be read in parallel
+    down to the individual records, is called "perfectly splittable".
+    (2) a "block-compressed" file format such as ``avroio``, in which a block of
+    records has to be read as a whole, but different blocks can be read in
+    parallel. (3) an "unsplittable" input such as a cursor in a database.
+
+    consumed number of split points:
+
+      * Before a source read (``BoundedSource.read()`` invocation) claims the
+        first split point, number of consumed split points is 0. This condition
+        holds independent of whether the input is splittable.
+      * Any source read that has only claimed one split point for which
+        ``RangeTracker.set_done()`` has not been invoked has 0 consumed split
+        points: the first split point is the current split point and is still
+        being processed. This condition holds independent of whether the input
+        is splittable.
+      * For an empty source read (which never invokes
+        ``RangeTracker.try_claim()``), the consumed number of split points is 0.
+        This condition holds independent of whether the input is splittable.
+      * For a non-empty, finished source read (in which
+        ``RangeTracker.try_claim()`` was invoked at least once and
+        ``RangeTracker.set_done()`` has been invoked), the value returned must
+        be at least 1 and should equal the total number of split points in the
+        source.
+      * For example (1): After returning record #30 (starting at 1) out of 50 in
+        a perfectly splittable 50-record input, this value should be 29. When
+        finished, the number of split points consumed should be 50.
+      * For example (2): In a block-compressed value consisting of 5 blocks, the
```

Comment:
This isn't correct. It stays at 0 until the first record of the second block is *claimed*. Remember, the range tracker never sees what gets returned.